File: code/dotnet/PredictorLibrary/Model4/Outputs/Model4Prediction.cs
namespace PredictorLibrary.Model4.Outputs;

public record Band(decimal Lower, decimal Upper, double Percentage);

public class Model4Prediction
{
    // Summary data will change, this is just so we can get something up on the screen

    public decimal Mean { get; set; }
    public decimal Minimum { get; set; }
    public decimal Maximum { get; set; }
    public List<Band> Bands { get; set; } = new();
    public List<Band> CumulativeBands { get; set; } = new();
    
    public List<Model4Iteration> Iterations { get; set; } = new();
}

File: code/angular/src/app/model4/models/Model4Prediction.ts
import { Model4Iteration } from "./Model4Iteration";

export class Model4Prediction {

  mean!: number;
  minimum!: number;
  maximum!: number;

  iterations!: Model4Iteration[];
  percentiles!: Percentile[];
  bands!: Band[];
  cumulativeBands!: Band[];
}

export class Percentile {
  percentileSelector!: number;
  value!: number;
}

export class Band {
  lower!: number;
  upper!: number;
  percentage!: number;
}
